mkt/webapps/tasks.py

```diff
@@ -744,3 +744,61 @@ def populate_is_offline(ids, **kw):
     for webapp in Webapp.objects.filter(pk__in=ids).iterator():
         if webapp.guess_is_offline():
             webapp.update(is_offline=True)
+
+
+@task
+@write
+def adjust_categories(ids, **kw):
+    ADJUSTMENTS = {
+        'books': 'books-comics',
+        'news-weather': 'news',
+    }
+    NEW_CAT_APPS = {
+        'food-drink': [424184, 439994, 442842, 444056, 444070, 444222, 444694,
+                       454558, 455620, 459304, 465445, 465700, 467828, 469104,
+                       470145, 471349, 476155, 477015, 497282, 500359, 501249,
+                       501573, 504143, 506111, 506691, 507921, 508211, 508215,
+                       508990],
+        'humor': [506369, 509746, 509848, 511390, 511504],
+        'internet': [488424, 489052, 499644, 500651, 505043, 505407, 505949,
+                     508828, 508830, 509160, 509606, 509722, 510114],
+        'kids': [364752, 364941, 449560, 466557, 466811, 473532, 473620,
+                 473865, 500527, 502263, 507497, 508089, 508229, 508239,
+                 508247, 509404, 509464, 509468, 509470, 509472, 509474,
+                 509476, 509478, 509484, 509486, 509488, 509490, 509492,
+                 509494, 509496, 509498, 509500, 509502, 509504, 509508,
+                 509512, 509538, 509540, 511502],
+        'personalization': [377564, 451302, 452888, 466637, 477186, 477304,
+                            477314, 480489, 480495, 481512, 482162, 488892,
+                            500037, 500041, 506495, 506581],
+        'science-tech': [367693, 424272, 460891, 468278, 468406, 469765,
+                         480750, 502187, 504637, 506187, 508672, 510050,
+                         511370, 511376, 512174, 512194],
+        'weather': [425986, 444314, 445008, 450602, 455256, 455660, 459364,
+                    461279, 461371, 462257, 463108, 466698, 468173, 470946,
+                    482869, 482961, 496946, 499699, 501553, 501581, 501583,
+                    502171, 502173, 502685, 503765, 505437, 506317, 506543,
+                    506553, 506623, 507091, 507139, 509150, 510118, 510334,
+                    510726, 511364],
+    }
+    # Adjust apps whose categories have changed.
+    for chunk in chunked(ids, 100):
+        for app in Webapp.objects.filter(pk__in=chunk):
+            save = False
+            for k, v in ADJUSTMENTS.items():
+                if k in app.categories:
+                    save = True
+                    app.categories.remove(k)
+                    app.categories.append(v)
+            if save:
+                task_log.info(u'[app:{0}] Adjusted categories: {1}'
+                              .format(app, app.categories))
+                app.save()
+    # Add apps to new categories.
+    for cat, apps in NEW_CAT_APPS.items():
+        for app in Webapp.objects.filter(pk__in=apps):
+            if cat not in app.categories:
+                app.categories.append(cat)
```

Comment:
Yeah, this is something I will fix. I have the above list but now in `{<app>: [<list of cats]}` format for that case.